sdk/search/azure-search-index/design.md

```diff
@@ -0,0 +1,403 @@
+        **kwargs
+    ) -> List[IndexingResult]:
+
+
+    def delete_documents(
+        self,
+        documents: List[dict],
+        **kwargs
+    ) -> List[IndexingResult]:
+
+
+    def merge_documents(
+        self,
+        documents: List[dict],
+        **kwargs
+    ) -> List[IndexingResult]:
+
+
+    def merge_or_upload_documents(
+        self,
+        documents: List[dict],
+        **kwargs
+    ) -> List[IndexingResult]:
+
+    def batch_update(
+        self,
+        batch: IndexOperationBatch,
+        **kwargs
+    ) -> List[IndexingResult]:
+```
+
+### `SearchQuery` API
+
+In addition to simple text searches, Azure Search supports more sophisticated
+search requests. These include ODATA queries for filtering, selecting, and
+ordering result and Lucene queries for search.
+
+```python
+azure.search.SearchQuery(search_text: str, **kwargs)
+
+class SearchQuery:
+    def filter(expression:str) -> None:
+    def orderby(*fields: List[str]) -> None:
+    def select(*fields: List[str]) -> None:
+```
+
+## Scenarios
+
+### 1. Get the number of documents in an Azure search index
+
+```python
+from azure.search import SearchApiKeyCredential, SearchIndexClient
+
+client = SearchIndexClient(
+    search_service_name="my-search-service",
+    index_name="hotels-sample-index",
+    credential=SearchApiKeyCredential(api_key="xxxxxxxxx")
+)
+
+print(client.get_document_count())
+```
+
+### 2. Get a specific document in an Azure search index
+
+```python
+from azure.search import SearchApiKeyCredential, SearchIndexClient
+
+client = SearchIndexClient(
+    search_service_name="my-search-service",
+    index_name="hotels-sample-index",
+    credential=SearchApiKeyCredential(api_key="xxxxxxxxx")
+)
+
+result = client.get_document(key="10")
+
+print(result)
+```
+
+### 3. Search for documents in an Azure search index with simple text
+
+```python
+from azure.search import SearchApiKeyCredential, SearchIndexClient
+
+client = SearchIndexClient(
+    search_service_name="my-search-service",
+    index_name="hotels-sample-index",
+    credential=SearchApiKeyCredential(api_key="xxxxxxxxx")
+)
+
+results = client.get_search_results(search_text="WiFi")
+
+for doc in results:
+    print(doc['Description'])
+    print(doc['Type'])
+    print(doc['BaseRate'])
+```
+
+### 3. Filter results Azure search index
+
+```python
+from azure.search import SearchApiKeyCredential, SearchIndexClient, SearchQuery
+
+client = SearchIndexClient(
+    search_service_name="my-search-service",
+    index_name="hotels-sample-index",
+    credential=SearchApiKeyCredential(api_key="xxxxxxxxx")
+)
+
+query = SearchQuery(
+    search_text="WiFi",
+    highlightPreTag="<em>",
+    highlightPostTag="</em>")
+
+# Base API accepts strings, templating/substitution TBD
+query.filter("Address/City eq 'Portland and Address/Country eq 'US'"))
+query.orderby("Rating desc", "geo.distance(FOO, geography'POINT(10 20)') asc")
+query.select("HotelId", "HotelName", "Rating")
```

Comment:
@johanste @brjohnstmsft I've removed the ODATA helper API. Note I would still advise the minimal convenience of letting `orderby` and `select` accept both a single string:
```
query.select("HotelId,HotelName,Rating")
```
And also accept a list of individual fields:
```
query.select("HotelId", "HotelName", "Rating")
```

Reply:
Sounds good to me.

Not sure this comparison matters, but for .NET Track 1 and Java Track 2 the methods/properties for orderby and select accept only lists of strings, not a single string.